gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/ProductDependency.groovy

```diff
@@ -15,8 +15,8 @@ class ProductDependency {
 
     ProductDependency() {}
 
-    ProductDependency(String productGroup, String productName, @Nullable String minimumVersion,
-                      @Nullable String maximumVersion, @Nullable String recommendedVersion) {
+    ProductDependency(String productGroup, String productName, String minimumVersion,
```

Comment:
Nice catch. These are still in-fact optional.  The just can't *both* be optional at the same time because that would technically make min and max version equal. I could hack around that specific case in this PR, but I figured it wasn't worth it since min version will become mandatory in #237